mmcv/transforms/wrappers.py

```diff
@@ -0,0 +1,447 @@
+        >>>             dict(type='Normalize'),
+        >>>         ])
+        >>> ]
+        >>> # Example 2:
+        >>> pipeline = [
+        >>>     dict(type='LoadImageFromFile', key='lq'),  # low-quality img
+        >>>     dict(type='LoadImageFromFile', key='gt'),  # ground-truth img
+        >>>     # ApplyToMultiple maps multiple outer fields to standard the
+        >>>     # inner field and process them with wrapped transforms
+        >>>     # respectively
+        >>>     dict(type='ApplyToMultiple',
+        >>>         # case 2: from one outer field that contains multiple
+        >>>         # data elements (e.g. a list)
+        >>>         # input_mapping=dict(img='images'),
+        >>>         inplace=True,
+        >>>         share_random_param=True,
+        >>>         transforms=[
+        >>>             dict(type='Crop', crop_size=(384, 384)),
+        >>>             dict(type='Normalize'),
+        >>>         ])
+        >>> ]
+    """
+
+    def __init__(self,
+                 transforms: List[Union[Dict, Callable[[Dict], Dict]]],
+                 input_mapping: Optional[Dict] = None,
+                 output_mapping: Optional[Dict] = None,
+                 inplace: bool = False,
+                 strict: bool = True,
+                 share_random_params: bool = False):
+        super().__init__(transforms, input_mapping, output_mapping, inplace,
+                         strict)
+
+        self.share_random_params = share_random_params
+
+    def scatter_sequence(self, data: Dict) -> List[Dict]:
+        # infer split number from input
+        seq_len = 0
+        key_rep = None
+        for key in self.input_mapping.keys():
+
+            assert isinstance(data[key], Sequence)
+            if seq_len:
+                if len(data[key]) != seq_len:
+                    raise ValueError('Got inconsistent sequence length: '
+                                     f'{seq_len} ({key_rep}) vs. '
+                                     f'{len(data[key])} ({key})')
+            else:
+                seq_len = len(data[key])
+                key_rep = key
+
+        if not seq_len:
+            raise RuntimeError(
+                'Fail to infer the sequence length. Please ensure that '
+                'the input items are sequences with the same length.')
+
+        scatters = []
+        for i in range(seq_len):
+            scatter = data.copy()
+            for key in self.input_mapping.keys():
+                scatter[key] = data[key][i]
+            scatters.append(scatter)
+        return scatters
+
+    def transform(self, results: Dict):
+        # Apply input remapping
+        inputs = self.remap_input(results, self.input_mapping)
+
+        # Scatter sequential inputs into a list
+        inputs = self.scatter_sequence(inputs)
+
+        # Control random parameter sharing with a contextmanager
+        if self.share_random_params:
+            cm = cache_random_params
```

Comment:
cm is not a clear abbreviation. Can we use a better name like cxt and add comments to explain it?